### PR TITLE
verify stack traces in thread sampling feature

### PR DESCRIPTION
**Pull request by Kielek**
## Why & What

Verify content of Sampled Stack Traces.

The application always return the same stack traces, so the verify functionality is the most reasonable way how to handle it.

## Tests

CI

# Notes
Extends #360 and #363

### CODE REVIEW

**Review by pjanotti — APPROVED** [review]
Excellent!

File: tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ThreadSamplingTests.cs [hunk 1]
```diff
@@ -32,8 +34,15 @@ public void SubmitTheadSamples()
...
                 var logsData = logsCollector.LogsData.ToArray();
                 // The application works for 5 seconds with debug logging enabled we expect at least 2 attempts of thread sampling in CI.
-                // On a dev box it is typical to get at least 3 but the CI machines seem slower, using 2 until the test is improved.
+                // On a dev box it is typical to get at least 3 but the CI machines seem slower, using 2
                 logsData.Length.Should().BeGreaterOrEqualTo(expected: 2);
+
+                var settings = VerifyHelper.GetThreadSamplingVerifierSettings();
+                foreach (var data in logsData)
+                {
+                    // all samples should be the same, as the testing application is just Thread.Sleep()ing
```
Comment (Kielek) [inline tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ThreadSamplingTests.cs:43]: My assumption was too optimistic.
Based on what I see in CI tests, stack traces vary on different env.
Manual verification will be needed. It is a plan on Monday.

Comment (Kielek) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ThreadSamplingTests.cs]: Fixed.
@pjanotti, please check if you are fine with current solution.

Comment (pjanotti) [reply tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/ThreadSamplingTests.cs]: It seems good @Kielek

**Pull request merged**